crates/wasm-mutate-stats/src/bin/wasm-mutate-stats.rs

```diff
@@ -0,0 +1,327 @@
+                            .collect(),
+                        errors: Vec::new(),
+                    };
+                    let serialized = serde_json::to_string(&newreport).unwrap();
+                    f.write(serialized.as_bytes())?;
+                    Ok(newreport)
+                })
+            })
+    }
+
+    fn save_report(&self, report: &Report, name: &String) {
+        let created = File::create(format!("{}.report.json", name));
+        created
+            .and_then(|mut f| {
+                let serialized = serde_json::to_string(report).unwrap();
+                f.write(serialized.as_bytes())
+            })
+            .expect("Report could not be written");
+    }
+
+    fn run_worker(&self, start: usize, end: usize) {
+        let mut wasmmutate = WasmMutate::default();
+        let mut corpus = self.corpus[start..end].to_vec();
+
+        loop {
+            match corpus.pop() {
+                Some((name, data)) => {
+                    println!("Wasm input {}", name);
+                    let mut wasm = data;
+
+                    let mut report = self.load_report(&name, wasm.clone()).unwrap();
+
+                    wasmmutate.seed(report.seed);
+                    let mut rng = SmallRng::seed_from_u64(report.seed);
+                    // mutate the initial wasm as many rotations are in the loaded report
+                    print!("Reaching previous state...");
+                    for _ in 0..report.rotations {
+                        match wasmmutate.run_with_rng(&wasm, &mut rng) {
+                            Ok(mutated) => {
+                                wasm = mutated;
+                            }
+                            Err(_) => { /* bypass */ }
+                        }
+                    }
+                    println!("Reached");
+                    while report.rotations <= ROTATIONS {
+                        report.rotations += 1;
+                        match wasmmutate.run_with_rng(&wasm, &mut rng) {
+                            Ok(mutated) => {
+                                let mut validator = wasmparser::Validator::new();
+                                match validator.validate_all(&mutated) {
+                                    Ok(_) => {
+                                        match self.compile(&mutated, &name, report.rotations) {
+                                            Ok(mhashes) => {
+                                                let hash = self.hash(mutated.clone());
+                                                wasm = mutated;
+                                                report.add_wasm_hash(hash);
+
+                                                for (idx, mhash) in mhashes.iter().enumerate() {
+                                                    report.add_obj_hash(mhash.clone(), idx);
+                                                }
+
+                                                if report.rotations % 100 == 0 {
```

Comment:
```suggestion
                                                if report.rotations % 100 == 99 {
```

So that we don't print a basically-empty report on the first iteration.

But backing up a bit: why do we print out / save to disk multiple reports? Why not just one at the end? I think this would also let us avoid the loading/deserializing of reports and make the code that much simpler.